jc/parsers/amixer.py

```diff
@@ -0,0 +1,205 @@
+from typing import List, Dict
+
+import jc.utils
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1'
+    description = '`amixer` command parser'
+    author = 'Eden Refael'
+    author_email = 'edenraf@hotmail.com'
+    compatible = ['linux']
+    magic_commands = ['amixer']
+    tags = ['command']
+
+
+__version__ = info.version
+
+def _process(proc_data: dict) -> dict:
+    """
+    Parameters:
+
+        proc_data:   (List of Dictionaries) raw structured data to process
+
+    Returns:
+
+        dictionary of amixer sget <control_name>
+    """
+    return proc_data
+
+
+def parse(
+    data: str,
+    raw: bool = False,
+    quiet: bool = False
+) -> List[Dict]:
+    """
+    Main text parsing function, The amixer is alsa mixer tool and output, Will work with Linux OS only.
+
+
+    The Algorithm for parsing the `amixer sget` command, Input Explained/Rules/Pseudo Algorithm:
+    1. There will always be the first line which tells the user about the control name.
+    2. There will always be the Capabilities which include many of capabilities - It will be listed and separated by `" "`.
+    3. After that we'll need to distinct between the Channel - Could be many of channels - It will be listed and separated
+       by `" "`.
+        3a. Capture channels - List of channels
+        3b. Playback channels - List of channels
+    4. Limits - We'll always have the minimum limit and the maximum limit.
+
+
+    Input Example:
+        1."":~$ amixer sget Capture
+        Simple mixer control 'Capture',0
+          Capabilities: cvolume cswitch
+          Capture channels: Front Left - Front Right
+          Limits: Capture 0 - 63
+          Front Left: Capture 63 [100%] [30.00dB] [on]
+          Front Right: Capture 63 [100%] [30.00dB] [on]
+
+
+
+
+        2."":~$ amixer sget Master
+        Simple mixer control 'Master',0
+          Capabilities: pvolume pvolume-joined pswitch pswitch-joined
+          Playback channels: Mono
+          Limits: Playback 0 - 87
+          Mono: Playback 87 [100%] [0.00dB] [on]
+
+
+
+
+
+        3."":~$ amixer sget Speaker
+        Simple mixer control 'Speaker',0
+          Capabilities: pvolume pswitch
+          Playback channels: Front Left - Front Right
+          Limits: Playback 0 - 87
+          Mono:
+          Front Left: Playback 87 [100%] [0.00dB] [on]
+          Front Right: Playback 87 [100%] [0.00dB] [on]
+
+
+
+
+        4."":~$ amixer sget Headphone
+        Simple mixer control 'Headphone',0
+          Capabilities: pvolume pswitch
+          Playback channels: Front Left - Front Right
+          Limits: Playback 0 - 87
+          Mono:
+          Front Left: Playback 0 [0%] [-65.25dB] [off]
+          Front Right: Playback 0 [0%] [-65.25dB] [off]
```

Comment:
Could you put this information below the `parse()` docstring? Otherwise this will all get pulled into the documentation. You can put this into regular commented lines or use another triple quote string below the docstring.

Reply:
so you meant like:
```python
def parse(
    data: str,
    raw: bool = False,
    quiet: bool = False
) -> List[Dict]:
"""
some docstring
"""

"""
another information.......
"""
```
?

Reply:
Did that, Resolved.